### PR TITLE
Fix source member lines from being truncated

### PR DESCRIPTION
**Pull request by richardm90**
### Changes

The record length of the source member was not always being respected and there were circumstances when the default record length of 80 characters was being assumed, losing characters between position 81 and 100.

See issue #1235 for full details.

When saving a source member if the record length has not been established then it will be determined before the save. The logic to establish the record length has been extracted into a private function called `getRecordLength` within `extendedContest.ts`, as this is used in two places.

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [x] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [x] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

File: src/filesystems/qsys/extendedContent.ts [hunk 1]
```diff
@@ -46,17 +46,8 @@ export class ExtendedIBMiContent {
       } catch (e) { }
 
       if (!this.sourceDateHandler.recordLengths.has(alias)) {
```
Comment (worksofliam) [inline src/filesystems/qsys/extendedContent.ts:48]: It looks like this is **always** returning false, even if the alias does exist in it.

Comment (worksofliam) [reply src/filesystems/qsys/extendedContent.ts:48]: Ok. My mistake, looks like we are internally deleting the stored source length in `sourceDateHandler.ts:124`.

**Comment by worksofliam** [comment]
@richardm90 Looks good and my testing looks good as well. Fancy fixing this conflict, and then we can merge!

**Comment by richardm90** [comment]
@worksofliam, I've fixed the conflict.

**Review by worksofliam — APPROVED** [review]
Thanks @richardm90! Thanks so much for the PR!

**Pull request merged**